code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -825,6 +825,39 @@
 	L.regenerate_icons()
 	qdel(src)
 
+/obj/item/slimepotion/slime/renaming
+	name = "renaming potion"
+	desc = "A potion that allows a self-aware being to change what name it subconciously presents to the world."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "potgreen"
+
+	var/being_used = FALSE
+
+/obj/item/slimepotion/slime/renaming/attack(mob/living/M, mob/user)
+	if(being_used || !ismob(M))
+		return
+	if(!M.ckey) //only works on animals that aren't player controlled
+		to_chat(user, "<span class='warning'>[M] is not self aware, and cannot pick its own name.</span>")
+		return
+
+	being_used = TRUE
+
+	var/new_name = stripped_input(M, "What would you like your name to be?", "Input a name", M.real_name, MAX_NAME_LEN)
+
+	if(!new_name || QDELETED(src) || QDELETED(M) || new_name == M.real_name)
+		being_used = FALSE
+		return
+
+	M.visible_message("<span class='notice'><span class='name'>[M]</span> has a new name, <span class='name'>[new_name]</span>.</span>", "<span class='notice'>Your old name of <span class='name'>[M.real_name]</span> fades away, and your new name <span class='name'>[new_name]</span> anchors itself in your mind.</span>")
+	message_admins("[ADMIN_LOOKUPFLW(M)] used [src] to rename themselves from <span class='name'>[M.real_name]</span> to <span class='name'>[new_name]</span>.")
```

Comment:
themselves?

Reply:
Well, the name is chosen by the person that the potion is affecting, not the person who used the potion.

Reply:
A clearer wording might be "[user] used [src] on [target], letting them rename themselves into [newname]."